### PR TITLE
lep service injection framework

### PR DESCRIPTION
**Pull request by sergeysenja1992**

### CODE REVIEW

File: xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/api/BaseLepContext.java [hunk 1]
```diff
@@ -16,6 +18,7 @@ public abstract class BaseLepContext {
     public BaseProceedingLep lep;
     public LepThreadHelper thread;
     public XmAuthenticationContext authContext;
+    public XmAuthenticationContextHolder authContextHolder;
```
Comment (RomanTeren) [inline xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/api/BaseLepContext.java:21]: is it okay that this value is not explicitly initialized anywhere? 

Shouldn't it be initialized in LepContextServiceImpl#createLepContext like this? 

```java
baseLepContext.authContextHolder = xmAuthContextHolder;
```

File: xm-commons-lep-groovy/src/main/java/com/icthh/xm/commons/lep/groovy/annotation/LepServiceTransformation.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    private Expression createExpression(String expression) {
+        String[] parts = expression.split("\\.");
+        Expression result = new VariableExpression(parts[0]);
+        for (int i = 1; i < parts.length; i++) {
+            result = new PropertyExpression(result, parts[i]);
+        }
+        return result;
+    }
+
+    private static boolean isLepServiceFactoryEnabled(ASTNode[] nodes, ClassNode classNode, SourceUnit source) {
```
Comment (RomanTeren) [inline xm-commons-lep-groovy/src/main/java/com/icthh/xm/commons/lep/groovy/annotation/LepServiceTransformation.java:212]: `classNode`, `source` are not used

**Review by RomanTeren — APPROVED** [review]

File: xm-commons-lep-groovy/src/main/java/com/icthh/xm/commons/lep/groovy/annotation/LepServiceTransformation.java (hunk 2 above)
Comment (sergeysenja1992) [reply xm-commons-lep-groovy/src/main/java/com/icthh/xm/commons/lep/groovy/annotation/LepServiceTransformation.java]: fixed

File: xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/api/BaseLepContext.java (hunk 1 above)
Comment (sergeysenja1992) [reply xm-commons-lep/src/main/java/com/icthh/xm/commons/lep/api/BaseLepContext.java:21]: fixed

**Review by pkev777 — APPROVED** [review]

**Pull request merged**